### PR TITLE
Update Select files and settings to sync.md

### PR DESCRIPTION
**Pull request by ryanjamurphy**
Trying to clarify this. People are still getting caught up with settings sync, unfortunately.

### CODE REVIEW

File: en/Obsidian Sync/Select files and settings to sync.md [hunk 1]
```diff
@@ -6,10 +6,15 @@ Any files or settings that have been synced to your [[Local and remote vaults|re
...
+## Sync vault settings and configurations and plugin settings and configurations
 
-1. Open **Settings** -> **Sync**.
+1. On your primary device, open **Settings** -> **Sync**.
 2. Under **Vault configuration sync**, enable the settings you want to sync.
+3. Wait for the settings to sync to your remote vault.
+4. On your secondary devices, open **Settings** -> **Sync**.
+5. Under **Vault configuration sync**, enable the settings you want to sync.
+6. Wait for the changes to download from your remote vault.
+7. Reload the app so that your synced settings take effect.
```
Comment (claremacrae) [inline en/Obsidian Sync/Select files and settings to sync.md:17]: Huge thanks for this great improvement.

It's great that this says 'Reload the app so that your synced settings take effect.'

There are then several following lists of settings changes to make (obviously pre-existing this commit)... It would be great if the same statement could be added to the others too.

And people should be told that every time they change any setting on any device, in any core or community plugin, or star/unstar a file, they should force-quit Obsidian every time, as a reflex.

File: en/Obsidian Sync/Select files and settings to sync.md [hunk 2]
```diff
@@ -6,10 +6,15 @@ Any files or settings that have been synced to your [[Local and remote vaults|re
 - Obsidian only applies vault settings during start-up. If you change a setting on one device, you need to restart Obsidian on your other devices for the new changes to take effect. For example, if you change the path of your daily notes in the [[Daily Notes]] plugin, you need to restart Obsidian on your other devices to use the new path.
 - ==Sync doesn't sync Sync's settings.== This allows users to configure Sync differently on each device according to their needs. This means, however, that you must configure custom Sync settings on each device.
 
-## Sync vault configuration
+## Sync vault settings and configurations and plugin settings and configurations
```
Comment (claremacrae) [inline en/Obsidian Sync/Select files and settings to sync.md:9]: Perhaps mention themes too?

File: en/Obsidian Sync/Select files and settings to sync.md (hunk 1 above)
Comment (marcusolsson) [reply en/Obsidian Sync/Select files and settings to sync.md:17]: LGTM!

Comment (marcusolsson) [reply en/Obsidian Sync/Select files and settings to sync.md:17]: Sorry for the confusion, I referred to the change, not your comment.

If restarting Obsidian is necessary for a change to apply, then it should definitely be added as a step.

> they should force-quit Obsidian every time, as a reflex

I don't think it would be helpful to add a general suggestion to restart Obsidian whenever something's not working (I'm not sure if that's what you meant). If there's a specific issue that may occur when syncing between multiple devices, then we should consider adding it to [Troubleshoot Obsidian Sync](https://help.obsidian.md/Obsidian+Sync/Troubleshoot+Obsidian+Sync).

Comment (claremacrae) [reply en/Obsidian Sync/Select files and settings to sync.md:17]: Hi Marcus, thank you for asking.

> If restarting Obsidian is necessary for a change to apply, then it should definitely be added as a step.

It is, but I hope the below explains why I feel making the restart a reflex is the most effective strategy. 😄 

> I don't think it would be helpful to add a general suggestion to restart Obsidian whenever something's not working (I'm not sure if that's what you meant).

No, I meant that if you enable syncing of any settings between devices, you should restart all other running instances of Obsidian in order for any changes in synced settings to take effect.

> If there's a specific issue that may occur when syncing between multiple devices, then we should consider adding it to [Troubleshoot Obsidian Sync](https://help.obsidian.md/Obsidian+Sync/Troubleshoot+Obsidian+Sync).

I feel that what follows is rather more general than for Troubleshooting.

I feel it is about sharing important information about a fundamental limitation in Obsidian Sync's behaviour (or Obsidian's behaviour - either way).


---

When first setting up sync, I probably lost around 10 to 15 hours to the types of problems listed below - and phrasing questions on Discord to try and get help - so I apologise in advance if this comes over as a little frustrated - it is not personal!

Here, approximately, are some concerns...

- It used to be (and I believe still is) a fundamental behaviour/limitation of Obsidian that it only reads any settings of any kind (application, core plugins, community plugins, settings for community plugins, themes...) when the application starts up.
- So long as you only have Obsidian running on one machine, that is absolutely not a problem
- But as soon as you use sync, you have the possibility of all sorts of things to happen
- At best these are annoying and confusing - so time-wasting...
- At worst, they lose data

The undocumented workaround in all cases is to restart Obsidian and make it load the latest settings.

And if you are on a mobile, you have to know that this means force-quitting (in Apple parlance)

I am hoping this this is what is covered by 'Sync: Better conflict resolution' in the Long-term column on the [Roadmap](https://trello.com/b/Psqfqp7I/obsidian-roadmap).

But until then, I feel it is vital that these pitfalls are explained in the docs, especially for a service that people are paying for.

Some examples... These are all specific examples of general categories.

## Plugin settings seemingly not getting copied over

- You turned on `Sync installed community plugins` (and their settings)
- But every time you change a setting in one machine, it doesn't take effect on your iPhone or iPad.
- So Sync of settings must be broken right?
- Or maybe the plugin is broken?
- But it happens for many plugins...
- But if you happen to know how to force-quit Obsidian on iPhone and iPad, you can work around it by restarting, to pick up the new settings.
- But what if you haven't used the device for a day or two?
- Can you remember whether you changed any settings on your desktop in that time?

## Problems syncing non-markdown files

- You decide you want to start syncing non-markdown content, say for a few Excel files
- On `device a`
    - You see you need to turn on `Detect all file extensions` in `Files and Links` settings
    - You turn on `Sync all other file types`
- On `device b`, nothing appears
    - You expected it would, because you had told Sync to sync your settings
- So you eventually turn on `Sync all other file types`
- Still nothing appears.
- Some time later, you discover that `Detect all file extensions` wasn't turned on on `device b` because Obsidian had not restarted.

## Stars get lost

1. You star `file 1` on `device a`. 
    1. It adds the file to the end of  `starred.json`
    2. `starred.json` is synced to `device b`
2. You star `file 2` on `device b`
    1. Obsidian on `device b` hasn't restarted, so doesn't know about the starring of `file 1` - 
    2. So when it saves the starring of `file 2`, it does so by overwriting the `file 1` line.
4. Back on `device a`, you add another star, or perhaps re-order some stars, 
    1. and you don't notice that `file 1` is no longer starred.

This happened to me when I did a load of starring of files and searches on Mac. It took ages and was rather tedious, especially because some items kept jumping to the bottom of the list when I tried re-ordering the stars... but I was pleased with the result.

Then I starred one note on my iPhone and all that effort was lost.

## Sync settings not getting copied over

- You know you turn on syncing of settings - and you made some changes to the settings of the Sync plugin
- It seems not to work
- So you restart Obsidian
- But it doesn't fix it
- Then you discover that the sync settings themselves are not syncronised
- (I assume this is not true, but have not tested recently)
- I think that this is because the Sync settings are not stored inside `.obsidian/`

**Thanks for reading! 🙏**

File: en/Obsidian Sync/Select files and settings to sync.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ericaxu) [reply en/Obsidian Sync/Select files and settings to sync.md:9]: Yeah, it's definitely too wordy
